internal/backend/scheduler/scheduler.go

```diff
@@ -0,0 +1,120 @@
+package scheduler
+
+import (
+	"context"
+	"fmt"
+
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+	"go.uber.org/zap"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/temporalclient"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+const (
+	taskQueueName = "scheduler-task-queue"
+	workerID      = "scheduler-worker"
+	workflowName  = "scheduler_workflow"
+)
+
+type Scheduler struct {
+	temporal   temporalclient.Client
+	sl         *zap.SugaredLogger
+	dispatcher sdkservices.Dispatcher
+}
+
+func New(l *zap.Logger, tc temporalclient.Client) *Scheduler {
+	return &Scheduler{sl: l.Sugar(), temporal: tc}
+}
+
+func (sch *Scheduler) Start(ctx context.Context, dispatcher sdkservices.Dispatcher) error {
+	sch.dispatcher = dispatcher
+
+	w := worker.New(sch.temporal.Temporal(), taskQueueName, worker.Options{Identity: workerID})
```

Comment:
didn't we say something about one worker that listen on all queues ?

Reply:
maybe in the future. not sure if it's a good idea.

Reply:
@efiShtain exactly. **Roughly** you can say that there should be a 1:1 relation between workers and TQs. Usually a Temporal worker is a separate binary, not bundled with other functionalities, and not mixed with other workers in the same binary. And that's why having a single worker responding to multiple TQs is either too many unnecessary TQs, or messy and risky coupling between workers.